### PR TITLE
Hide data insights and heatmap

### LINKED ISSUE #725: [Queue Refactor] - Disable data insights and heatmap
** do this on the `hybrid-queues` branch please!!

As we refactor the QueueModel, it's going to break our data insights and heatmap. We're gonna disable this stuff until 

Frontend:
- hide the data insights page and insights tab
- hide the heatmap

Backend:
- don't return heatmap/insights data (I'm not sure if you really need to disable the insights endpoints much, but you will need to avoid returning the heatmap with the rest of the GET course data. There might be some more investigation here on exactly how insights data is returned from the backend)

### PR DESCRIPTION
**Pull request by brianyyang**
# Description

As we refactor QueueModel, insights and heatmap might break so we disabled those on the front end and back end.

Closes #725 

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)

# How Has This Been Tested?

We didn't the insights.services file so the insights test suite still works properly. Instead we throw errors if anyone tries to reach the endpoints through the controller.

As for the front end, make sure the insights tab on the nav bar doesn't show up for professors. Also, if you manually navigate to the page through ./course/#/insights, make sure no back end errors are popping up and the page should tell you that insights are disabled.

# Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have commented my code where needed 
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by IrisLiu-00 — APPROVED** [review]
LGTM! Although I will defer to @dfarooq610 since he knows much more about data insights than I

File: packages/app/components/Nav/NavBar.tsx [hunk 1]
```diff
@@ -180,13 +180,16 @@ export default function NavBar({ courseId }: NavBarProps): ReactElement {
       text: "Queue",
     });
   }
+
+  /**
```
Comment (dfarooq610) [inline packages/app/components/Nav/NavBar.tsx:184]: Nit: just add a TODO here so we know to re-enable it eventually (maybe link this issue on GH)

File: packages/app/pages/course/[cid]/insights.tsx [hunk 2]
```diff
@@ -1,4 +1,5 @@
-import React, { ReactElement, useState } from "react";
+import React, { ReactElement } from "react";
```
Comment (dfarooq610) [inline packages/app/pages/course/[cid]/insights.tsx:1]: Instead of commenting this all out, can we just redirect to a 404 for a bit lmao?idk

**Review by dfarooq610 — CHANGES_REQUESTED** [review]
Something else you might want to disable is the generation of the heatmap. The heatmap service is within the course, and the easiest way to "disable" it is to just set it to the Boolean False when you get a course

File: packages/server/test/course.integration.ts [hunk 3]
```diff
@@ -401,7 +401,7 @@ describe('Course Integration', () => {
       });
 
       const thenThreeMoreHours = new Date(yesterdayPlusTwoHours);
-      thenThreeMoreHours.setUTCHours(yesterday.getUTCHours() + 3);
+      thenThreeMoreHours.setUTCHours(yesterdayPlusTwoHours.getUTCHours() + 3);
```
[the remark this answers is not in the document]
Comment (brianyyang) [reply packages/server/test/course.integration.ts:404]: yes

File: packages/app/components/Nav/NavBar.tsx (hunk 1 above)
Comment (brianyyang) [reply packages/app/components/Nav/NavBar.tsx:184]: good idea!

File: packages/app/pages/course/[cid]/insights.tsx (hunk 2 above)
Comment (brianyyang) [reply packages/app/pages/course/[cid]/insights.tsx:1]: that sounds like a plan

Comment (brianyyang) [reply packages/app/pages/course/[cid]/insights.tsx:1]: nevermind, we decided having the message saying insights is temporarily disabled would be a nice to have

**Review by dfarooq610 — APPROVED** [review]

**Pull request merged**